libs/vm/src/analyser.cpp

```diff
@@ -1182,21 +1274,38 @@ bool Analyser::AnnotateEqualityOp(ExpressionNodePtr const &node)
     bool const lhs_is_primitive = lhs->type->IsPrimitive();
     if (rhs_is_concrete_type)
     {
+      if (rhs->node_kind == NodeKind::InitializerList)
+      {
+        if (!ConvertInitializerList(rhs, lhs->type))
+        {
+          AddError(node->line, "incompatible types");
+          return false;
+        }
+      }
+      else if (lhs->node_kind == NodeKind::InitializerList)
+      {
+        if (!ConvertInitializerList(lhs, rhs->type))
+        {
+          AddError(node->line, "incompatible types");
+          return false;
+        }
+      }
+
```

Comment:
Just curious what happens here if LHS and RHS are both init lists?

Reply:
Compilation error, as current type handling is to narrow to contain proper matching of untyped lists.

Reply:
A compiler error is fine, because {1, 2, 3} op {1, 2, 3} (say) doesn't really make any sense when we can't infer the types.